robot/impl/local_robot.go

```diff
@@ -669,10 +674,19 @@ func (r *localRobot) updateWeakDependents(ctx context.Context) {
 	// For now, we pass all resources and empty configs.
 	for resName, res := range internalResources {
 		switch resName {
-		case web.InternalServiceName, framesystem.InternalServiceName:
+		case web.InternalServiceName:
 			if err := res.Reconfigure(ctx, allResources, resource.Config{}); err != nil {
 				r.Logger().Errorw("failed to reconfigure internal service", "service", resName, "error", err)
 			}
+		case framesystem.InternalServiceName:
+			fsCfg, err := r.FrameSystemConfig(ctx)
+			if err != nil {
+				r.Logger().Errorw("failed to reconfigure internal service", "service", resName, "error", err)
+				continue
+			}
+			if err := res.Reconfigure(ctx, components, resource.Config{ConvertedAttributes: fsCfg}); err != nil {
```

Comment:
Two things to note here that @raybjork and I discussed offline. 

With this change, the frame system service is weakly dependent on all _components_. That means changes in the frames of services will not cause a reconfiguration of the frame system service. If the MP team is planning on supporting services within the frame system, might be good to leave a `TODO` with a ticket number here to include services.

I also see that we're shoehorning the frame system config into `ConvertedAttributes`. We [also do this for remotes](https://github.com/viamrobotics/rdk/blob/main/robot/impl/resource_manager.go#L110), so there is precedence for it. I don't love it as a workaround to fit non-component/non-service objects into a `resource.Config`, but if no one has better alternatives, I think it's fine for now.